File: apps/omnimereduapp/lib/domain/usecases/grade/get_grade_by_id_usecase.dart
import '../../entities/grade/grade_entity.dart';
import '../../repositories/school/grade_repository.dart';

class GetGradeByIdUseCase {
  final GradeRepository repository;

  GetGradeByIdUseCase(this.repository);

  Future<GradeEntity> call(String id) async {
    return await repository.getGradeById(id);
  }
}

File: services/user_service/src/domain/usecases/grade/GetGradeByIdUseCase.ts
import { IGradeRepository } from "../../repositories/IGradeRepository";
import { Grade } from "../../entities/Grade";

export class GetGradeByIdUseCase {
  constructor(private gradeRepository: IGradeRepository) {}

  async execute(id: string): Promise<Grade | null> {
    return await this.gradeRepository.findById(id);
  }
}
